vaporetto/src/type_predictor.rs

```diff
@@ -0,0 +1,205 @@
+use crate::model::ScoreValue;
+use crate::sentence::Sentence;
+use daachorse::DoubleArrayAhoCorasick;
+
+pub enum TypePredictor {
```

Comment:
I think `TypeScorer` is better.
This name seems that the function predicts types.